source/common/tracing/zipkin/zipkin_core_types.cc

```diff
@@ -71,6 +71,14 @@ Annotation& Annotation::operator=(const Annotation& ann) {
   return *this;
 }
 
+void Annotation::changeEndpointServiceName(const std::string& service_name) {
+  if (endpoint_.valid()) {
+    Endpoint endpoint(endpoint_.value());
```

Comment:
Sorry just to clarify, when you add the new non-const version, have the const version call the non-const version to avoid code duplication. :)